### PR TITLE
[Feat] - log metadata on traces + allow users to log metadata when `existing_trace_id` exists 

### PR DESCRIPTION
**Pull request by ishaan-jaff**
## Type

🆕 New Feature
🐛 Bug Fix
🧹 Refactoring
📖 Documentation
💻 Development Environment
🚄 Infrastructure
✅ Test

## Pre-Submission Checklist (optional but appreciated):

- [ ] I have included relevant documentation updates (stored in /docs/my-website)

## OS Tests (optional but appreciated):

- [ ] Tested on Windows
- [ ] Tested on MacOS
- [ ] Tested on Linux

### CODE REVIEW

File: litellm/integrations/langfuse.py [hunk 1]
```diff
@@ -318,11 +318,11 @@ def _log_langfuse_v2(
                     else:
                         clean_metadata[key] = value
 
-            session_id = clean_metadata.pop("session_id", None)
-            trace_name = clean_metadata.pop("trace_name", None)
-            trace_id = clean_metadata.pop("trace_id", None)
-            existing_trace_id = clean_metadata.pop("existing_trace_id", None)
-            update_trace_keys = clean_metadata.pop("update_trace_keys", [])
+            session_id = clean_metadata.get("session_id", None)
```
Comment (ishaan-jaff) [inline litellm/integrations/langfuse.py]: cc @alexanderepstein some users mentioned wanting this data in langfuse. I believe you worked on a PR for this. Just sharing so you're aware of the change

File: litellm/integrations/langfuse.py [hunk 2]
```diff
@@ -342,6 +342,10 @@ def _log_langfuse_v2(
                         if updated_trace_value is not None:
                             trace_params[trace_param_key] = updated_trace_value
 
+                if "metadata" in update_trace_keys:
+                    # log metadata in the trace
+                    trace_params["metadata"] = clean_metadata
+
```
Comment (ishaan-jaff) [inline litellm/integrations/langfuse.py]: if a user wants to update the `metadata` in the trace

File: litellm/integrations/langfuse.py [hunk 3]
```diff
@@ -363,6 +367,7 @@ def _log_langfuse_v2(
                         "trace_version", clean_metadata.get("version", None)
                     ),  # If provided just version, it will applied to the trace as well, if applied a trace version it will take precedence
                     "user_id": user_id,
+                    "metadata": metadata,
                 }
```
Comment (ishaan-jaff) [inline litellm/integrations/langfuse.py]: for non existing traces - add metadata

**Comment by alexanderepstein** [comment]
I think the current code that is released already supports this, unless there is something I am missing. And the .get will cause duplication of data as anything not popped will also show up in the generations metadata.

 https://docs.litellm.ai/docs/observability/langfuse_integration#set-custom-trace-id-trace-user-id-trace-metadata-trace-version-trace-release-and-tags

**Comment by ishaan-jaff** [comment]
@alexanderepstein 
- some users are complaining about litellm not using the correct `trace_ids` so we want to land that in the trace metadata 

My goal is to show users the exact `metadata` passed in the call as a trace_param so they know what they are passing to litellm traces 

example curl:

```shell
curl --location 'http://localhost:4000/chat/completions' \
    --header 'Authorization: Bearer sk-MBJNDqRfEMl45kFqscFPFQ' \
    --header 'Content-Type: application/json' \
    --data '{
    "model": "gpt-3.5-turbo",
    "user": "litellm2",
    "metadata": {"trace_id": "test2", "existing_trace_id": "test5"},
    "messages": [
        {
        "role": "user",
        "content": "tes4"
        }
    ]
}'
```

**Comment by ishaan-jaff** [comment]
I don't see any duplication of data - see this trace: https://us.cloud.langfuse.com/project/clw1axtf000026l1qbpr6jvzk/traces/444e7a8f-4634-4411-9fe2-890ee28fdf14?observation=time-10-50-42-016734_chatcmpl-9Nl9mzKtUcsOeHyReL89usZJTXh6w 

How do I repro the scenario where data duplication would occur ?

**Comment by alexanderepstein** [comment]
I don't understand the example there, why does trace_id != existing_trace_id ? Maybe it's a use case I haven't thought of but I can't think of why you would want to do that.

**Comment by ishaan-jaff** [comment]
> I don't understand the example there, why does trace_id != existing_trace_id ? Maybe it's a use case I haven't thought of but I can't think of why you would want to do that.

even if they are equal - this data never lands as metadata in the trace  -> (`trace_id`, `existing_trace_id`)

since it does not land in the trace metadata users cannot debug why litellm is not using the expected `trace_id`

**Comment by alexanderepstein** [comment]
If this is just a debugging thing then I would not make it the default behavior, I don't think it makes sense to have a trace ID in langfuse and also in the metadata of the trace, it feels like unnecessary duplication. 

I'm also not sure why this is litellms problem to solve, can't the users just dump their metadata to console or some local file before making the litellm call? That should be sufficient for debugging.

**Comment by ishaan-jaff** [comment]
> If this is just a debugging thing then I would not make it the default behavior

I agree with you - i'll edit this PR to hide this behind a debug flag

**Pull request merged**